cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/runtime/workflow/WorkflowDriver.java

```diff
@@ -375,11 +375,11 @@ private void executeFork(final ApplicationSpecification appSpec, WorkflowForkNod
           String branchInfo = retValue.getKey();
           WorkflowToken branchToken = retValue.getValue();
           ((BasicWorkflowToken) token).mergeToken(branchToken);
-          LOG.info("Execution of branch {} for fork {} completed", branchInfo, fork);
+          LOG.trace("Execution of branch {} for fork {} completed", branchInfo, fork);
         } catch (Throwable t) {
           Throwable rootCause = Throwables.getRootCause(t);
           if (rootCause instanceof ExecutionException) {
-            LOG.error("Exception occurred in the execution of the fork node {}.", fork, rootCause);
+            LOG.error("Exception occurred in the execution of the fork node {}.", fork);
```

Comment:
This if statement will never be true, as `ExecutionException` can never be a root cause. Also, since the method already has `throws Exception`, the whole `catch` block should be like this:

```java
        } catch (InterruptedException e) {
          // Due to workflow abortion, so just break the loop and let finally block to cancel tasks
          break;
        } catch (ExecutionException e) {
          // Unwrap the cause
          throw e.getCause();
        }
```